src/eval/operation.rs

```diff
@@ -1955,6 +1955,42 @@ impl<R: ImportResolver> VirtualMachine<R> {
                     ))
                 }
             },
+            BinaryOp::DiffRecordFields() => match_sharedterm! {t1,
+                with {
+                    Term::Record(r1) => match_sharedterm! {t2,
+                        with {
+                            Term::Record(r2) => {
+                                let mut r1 = r1;
+                                r1.fields.retain(|k, _| !r2.fields.contains_key(k));
+                                Ok(Closure {
+                                    body: RichTerm::new(Term::Record(r1), fst_pos),
+                                    env: env1
+                                })
+                            }
+                        } else {
+                            Err(EvalError::TypeError(
+                                String::from("Record"),
+                                String::from("%record_diff_fields%, 2nd argument"),
+                                snd_pos,
+                                RichTerm {
+                                    term: t2,
+                                    pos: pos2
+                                }
+                            ))
+                        }
+                    }
+                } else {
+                    Err(EvalError::TypeError(
+                        String::from("Record"),
+                        String::from("%record_diff_fields%, 1st argument"),
+                        fst_pos,
+                        RichTerm {
+                            term: t1,
+                            pos: pos1,
+                        }
+                    ))
+                }
+            },
```

Comment:
Can we add tests for this case please? :pray: 

I'm thinking at least a few examples (unit tests), and I can think off the top of my head about the following PBTs:
* `for any record r1, diff_record_fields r1 r1 == {}`
* `for any record r1, any list of extra fields (except SPRT) extra_fields, diff_record_fields (r1 // extra_fields) r1 == diff_record_fields r1 (r1 // extra_fields) == extra_fields`
* `for any record r1, for any record r2, diff_record_fields r1 r2 (except SPRT) == diff_record_fields r2 r1 (except SPRT)`

Reply:
I'm not necessarily against adding these tests, but I'll point out that this op is already tested by virtue of its use inside the `$record` contract, so every test for that contract verifies that this is a valid change. 

In general, I'm not sure what the testing strategy is here. On the one hand, this is only ever intended to be used as an implementation detail inside the `$record` contract, so you could argue that it's sufficient to just rely on the tests for that feature. On the other hand, there's nothing (lack of documentation aside) to stop users from calling the `%record_diff_fields%` primitive in their own code, so I can see why we might want this to have its own suite of tests.

@yannham do you have any thoughts on this?

Reply:
I guess I'll add: my personal opinion is that we should prioritise testing via the (_intended_) public surface area of the language, so I'd be in favour of not adding tests here, given that any change to this code will break most of the tests for record contracts anyway.

Ultimately I'm happy to do whatever we all agree makes sense, but it'd be great to document whatever decision we reach for future reference.